src/hooks/tradingView/useChartLines.ts

```diff
@@ -189,59 +222,187 @@ export const useChartLines = ({
     });
   }, [stringGetter, currentMarketId, currentMarketPositionData, maybeDrawPositionLine]);
 
+  // Cache for order modification that stores the new orders that are submitted but not yet placed
+  const pendingOrderAdjustmentsRef = useRef<{
+    [clientId: string]: { orderPayload: HumanReadablePlaceOrderPayload; oldOrderId: string };
+  }>({});
+
+  const removePendingOrderAdjustment = (clientId: number) => {
+    const { [clientId]: removed, ...withoutOrderId } = pendingOrderAdjustmentsRef.current;
+    pendingOrderAdjustmentsRef.current = withoutOrderId;
+  };
+
+  const addPendingOrderAdjustment = (
+    orderPayload: HumanReadablePlaceOrderPayload,
+    oldOrderId: string
+  ) => {
+    pendingOrderAdjustmentsRef.current = {
+      ...pendingOrderAdjustmentsRef.current,
+      [orderPayload.clientId]: { orderPayload, oldOrderId },
+    };
+  };
+
+  const dispatch = useAppDispatch();
+
+  const onMoveOrderLine = useCallback(
+    async (order: SubaccountOrder, orderLine?: IOrderLineAdapter) => {
+      if (!orderLine || !canModifyOrderTypeFromChart(order)) return;
+
+      const oldPrice = order.triggerPrice ?? order.price;
+      const newPrice = orderLine.getPrice();
+
+      // TODO(tinaszheng): do validation here for new price
+      // make sure the newPrice doesnt cross over the current price depending
+      // on the direction of the trade
+
+      orderLine.setPrice(newPrice);
+      // Don't go through abacus for limit order modifications to avoid having to override any trade inputs in the Trade Form
+      const orderPayload = createPlaceOrderPayloadFromExistingOrder(order, newPrice);
+      if (!orderPayload) {
+        orderLine.setPrice(oldPrice);
+        return;
+      }
+
+      addPendingOrderAdjustment(orderPayload, order.id);
+
+      // Dispatch both actions here so that the user sees both cancel + submitting notifications together
+      dispatch(cancelOrderSubmitted(order.id));
+      dispatch(
+        placeOrderSubmitted({
+          marketId: orderPayload.marketId,
+          clientId: orderPayload.clientId,
+          orderType: orderPayload.type as TradeTypes,
+        })
+      );
+
+      const { success: cancelSuccess } = await cancelOrderAsync(order.id);
+      if (!cancelSuccess) {
+        dispatch(
+          cancelOrderFailed({
+            orderId: order.id,
+            errorParams: DEFAULT_SOMETHING_WENT_WRONG_ERROR_PARAMS,
+          })
+        );
+        dispatch(
+          placeOrderFailed({
+            clientId: orderPayload.clientId,
+            errorParams: DEFAULT_SOMETHING_WENT_WRONG_ERROR_PARAMS,
+          })
+        );
+        orderLine.setPrice(oldPrice);
+        removePendingOrderAdjustment(orderPayload.clientId);
+        return;
+      }
+
+      dispatch(cancelOrderConfirmed(order.id));
+
+      const res = await abacusStateManager.chainTransactions.placeOrderTransaction(orderPayload);
```

Comment:
is this the same as the `placeOrderTransaction` in `DydxChainTransactions`?

Reply:
yeah `chainTransactions` is the instance of `DydxChainTransactions` tied to the abacus state manager